modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1408,16 +1407,14 @@ export class Wallet implements IWallet {
       return params.prv;
     }
 
-    const userKeychain = (await this.getEncryptedUserKeychain()) as any;
-    const userEncryptedPrv = userKeychain.encryptedPrv;
-
-    let userPrv;
-    try {
-      userPrv = this.bitgo.decrypt({ input: userEncryptedPrv, password: params.walletPassphrase });
-    } catch (e) {
-      throw new Error('error decrypting wallet passphrase');
+    const userKeychain = await this.getEncryptedUserKeychain();
+    if (!params.walletPassphrase) {
+      throw new Error('wallet passphrase was not provided');
+    }
```

Comment:
```suggestion
```
Probably not needed since we check if the `prv` and `walletPassphrase` are undefined, and we return if the `prv` exist, meaning that the other case is that the `walletPassphrase` exists

Reply:
ah fair, this is redundant.

Reply:
Ah i remember why I added this now, there isn't enough type narrowing here for typescript to know that `walletPassphrase` is not undefined at this point. To add some better type narrowing requires some refactors that I didn't want to introduce at this point.